archive/overlap.go

```diff
@@ -0,0 +1,125 @@
+package archive
+
+import (
+	"sort"
+
+	"github.com/brimsec/zq/pkg/nano"
+	"github.com/brimsec/zq/zbuf"
+)
+
+// mergeChunksToSpans takes an unordered set of Chunks with possibly overlapping
+// spans, and returns an ordered list of SpanInfos, whose spans will be bounded
+// by filter, and where each spanInfo contains one or more Chunks whose data
+// falls into the spanInfo's span.
+func mergeChunksToSpans(chunks []Chunk, dir zbuf.Direction, filter nano.Span) []spanInfo {
```

Comment:
Nit: To avoid confusion with filter.Filter, maybe rename filter to span?

Reply:
I've gone back & forth on this: I'd like to keep it as filter, otherwise the comments above become really hard to read. ('the spaninfo's span is bounded by span').

Reply:
"... ordered list of spanInfos, each bounded by span and containing one or more ..."?

Reply:
But this is just a nit, so definitely feel free to leave it as is.